internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -88,6 +90,10 @@ func (r reconcileComplete) Dirty() bool {
 	return r.dirty
 }
 
+func (r reconcileComplete) Complete() bool {
+	return r.dirty || r.complete
```

Comment:
If this is the logic you want, why do we need both variables?

I'm also curious to understand why this is the desired logic.

Reply:
I think the naming I picked is confusing and I need to incorporate @rwsu  feedback. The logic I am after is having a way to signal when the `Reconcile` should be stopped, which can be in the case of a resource update or an explicit stop flag.

It could be argued that `dirty` should not imply a return but this is how we have used it throughout the whole Reconcile function so I wanted to incorporate that. It may be an optimization we can do way with.

Reply:
> It could be argued that dirty should not imply a return but this is how we have used it throughout the whole Reconcile function so I wanted to incorporate that.

What I [see on master](https://github.com/openshift/assisted-service/blob/b725f51/internal/controller/controllers/bmh_agent_controller.go#L148-L185) is that `dirty == true` causes an update, but the Reconcile logic then continues as long as there isn't an error performing the update.

Reply:
> What I see on master is that dirty == true causes an update, but the Reconcile logic then continues as long as there isn't an error performing the update.

Correct, that's for the reconcile steps that come after the `reconcileBMH`. I have removed this part and now `dirty` and `stop` are different. I will do another small change to make the usage of `stop` consistent in the `reconcileBMH` case.